distribution/lib/Standard/Visualization/0.0.0-dev/src/Table/Visualization.enso

```diff
@@ -191,28 +191,122 @@ make_json_for_xml_element xml_element max_items type:Text="XML_Element" =
-    has_index_col        = ["has_index_col", include_index_col]
+    has_index_col        = ["has_index_col", True]
+    is_ssrm              = if is_db_table then False else all_rows_count > max_rows
     links                = ["get_child_node_action", "get_row"]
     child_label          = ["child_label", "row"]
+    data                 = if is_ssrm then Nothing else columns.map get_vector
     data_quality_metrics = if is_db_table then [] else
         number_nothing = JS_Object.from_pairs [["name", "Count nothings"], ["percentage_value", columns.map .count_nothing]]
         sampled_label = if all_rows_count > Column.default_sample_size then " (sampled)" else ""
         number_untrimmed = JS_Object.from_pairs [["name", "Count untrimmed whitespace" + sampled_label], ["percentage_value", columns.map .count_untrimmed]]
         number_non_triv = JS_Object.from_pairs [["name", "Count non trivial whitespace" + sampled_label], ["percentage_value", columns.map .count_non_trivial_whitespace]]
         JS_Object.from_pairs
         [number_nothing, number_untrimmed, number_non_triv]
-    pairs                = [header, value_type, data, all_rows, has_index_col, links, ["data_quality_metrics", data_quality_metrics] ,["type", "Table"], child_label]
+    pairs                = [header, value_type, ["data", data], all_rows, has_index_col, links, ["data_quality_metrics", data_quality_metrics] ,["type", "Table"], child_label,  ["is_ssrm", is_ssrm]]
     JS_Object.from_pairs pairs
 
+## PRIVATE
+
+   applies the sort model to the table.
+
+   Arguments:
+   - start_number: the start index of the next 100 rows to get
+   - sort_col_index_list: indexes of columns with sorts applies
+   - sort_direction_list: list of sort directions 1 for ascending -1 for descending
+
+get_distinct_values_for_column : Table -> Integer -> JS_Object
+get_distinct_values_for_column x column_index =
+    column = x.at column_index
+    values = column.to_vector.distinct
+    result = JS_Object.from_pairs [["distinct_vals", values]]
+    result.to_text
+
+## PRIVATE
+
+   applies the sort model to the table.
+
+   Arguments:
+   - x: table to sort
+   - sort_col_index_list: indexes of columns with sorts applies
+   - sort_direction_list: list of sort directions 1 for ascending -1 for descending
+
+apply_sort_to_table : Table -> List -> List -> Table
+apply_sort_to_table x sort_col_index_list sort_direction_list =
+    parse_direction d = if d == -1 then Sort_Direction.Descending else Sort_Direction.Ascending
+    get_pattern i = (Sort_Column.Index (sort_col_index_list.get i) (parse_direction (sort_direction_list.get i)))
+    sort_pattern = ((0.up_to sort_col_index_list.length).map idx-> get_pattern idx)
+    x.sort sort_pattern
+
+## PRIVATE
+
+   applies the filter model to the table.
+
+   Arguments:
+   - table: table to sort
+   - filter_col: indexes of columns with sorts applies
+   - filter_action: list of filter actions 
```

Comment:
I'm a bit confused - the comment here says that this is a _list_ of actions but the type signature says `Filter_Condition` - indicating a single value not a list. Looking at the code it is not a list, so perhaps this comment became outdated?